File: seed/ruby-sdk-v2/exhaustive/wire-tests/lib/seed/endpoints/object/types/get_and_return_nested_with_required_field_object_request.rb
# frozen_string_literal: true

module Seed
  module Endpoints
    module Object_
      module Types
        class GetAndReturnNestedWithRequiredFieldObjectRequest < Internal::Types::Model
          field :string_value, -> { String }, optional: false, nullable: false, api_name: "stringValue"

          field :body, -> { Seed::Types::TypesNestedObjectWithRequiredField }, optional: false, nullable: false
        end
      end
    end
  end
end

File: seed/php-sdk/exhaustive/wire-tests/src/Endpoints/Object/Requests/GetAndReturnNestedWithRequiredFieldObjectRequest.php
<?php

namespace Seed\Endpoints\Object\Requests;

use Seed\Core\Json\JsonSerializableType;
use Seed\Types\TypesNestedObjectWithRequiredField;

class GetAndReturnNestedWithRequiredFieldObjectRequest extends JsonSerializableType
{
    /**
     * @var TypesNestedObjectWithRequiredField $body
     */
    public TypesNestedObjectWithRequiredField $body;

    /**
     * @param array{
     *   body: TypesNestedObjectWithRequiredField,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->body = $values['body'];
    }
}
